### ISSUE TITLE
The new backlinks panel is not referencing all notes

### ISSUE THREAD

**Issue by danielarmengolaltayo** [foambubble/foam#525]
- Foam version: 0.11.0
- Platform: Mac

**Summary**
The new [backlinks panel](https://github.com/foambubble/foam/pull/514) doesn't find references for .md files stored inside a folder with a name starting with "_" or "-" (just tested these two cases, but I assume that the issue extends to other symbols).

The "old" backlinks panel (third-party extension), works just fine in this case.

**Comment by riccardoferretti** [comment]
I have tested this and it works on my machine (classic) - can you add an example to help reproduce the issue?

**Comment by danielarmengolaltayo** [comment]
- Two files:
  - foam/inbox.md
  - foam/foo/bar.md
- First file, has a link to the second [[bar]]
- Backlinks panels, both new and old, working fine.

<img width="370" alt="01" src="https://user-images.githubusercontent.com/7153778/110943891-04fe8e80-833c-11eb-8937-06528079248b.png">

In the second screenshot you can see how the new backlinks panel loses the reference when changing the name of the folder "foo" to "_foo".

<img width="406" alt="02" src="https://user-images.githubusercontent.com/7153778/110943844-fa43f980-833b-11eb-878b-81f1469e2475.png">

You can see in the second screenshot how the folder changes its color. Same behavior happens with files placed inside foam/.foam/templates folder, for example. Maybe there is a rule somewhere to exclude files inside this type of folders and that's why you place it the "templates" folder inside the ".foam" folder. I don't know, just guessing. 🤷‍♂️

Tell me if you need something else or more specific to reproduce the issue.

**Comment by danielarmengolaltayo** [comment]
mmmm, sorry, I opened the issue in the wrong repo, I thought I was in the foam-template. 

And I tried to reproduce the issue in a fresh and clean repo I just cloned (foam-template), and the behavior is different. Maybe I messed the code somewhere but I can not find the source. I close the issue and I'll move all my notes to a new repo. I will open the issue again (in the correct repo) if I can reproduce the issue.

Thank you @riccardoferretti and sorry for the confusion.
